trade_management_unit/models/UserConfiguration.py

```diff
@@ -1,21 +1,203 @@
         db_table = "user_configurations"
+        verbose_name = "User Configuration"
+        verbose_name_plural = "User Configurations"
+        ordering = ['user_id']
+        constraints = [
+            models.CheckConstraint(
+                check=models.Q(min_reward_risk_ratio__lte=models.F('max_reward_risk_ratio')),
+                name='chk_reward_risk_ratio_order'
+            ),
+            models.CheckConstraint(
+                check=models.Q(default_stop_loss_percentage__lte=models.F('default_take_profit_percentage')),
+                name='chk_stop_loss_take_profit_order'
+            ),
+        ]
 
+    def __str__(self):
+        return f"Configuration for User {self.user_id}"
 
     @classmethod
     def get_attribute(cls, user_id, attribute):
+        """Legacy method for backward compatibility"""
         try:
-            user_config = cls.objects.get(user_id=user_id)
+            user_config = cls.objects.get(user_id=user_id, is_active=True)
             return getattr(user_config, attribute, None)
         except cls.DoesNotExist:
             return None
+
+    @classmethod
+    def get_or_create_for_user(cls, user_id, **kwargs):
+        """Get or create configuration for a user with default values"""
+        defaults = {
+            'risk_appetite': 5.0,
+            'min_reward_risk_ratio': 2.0,
+            'max_reward_risk_ratio': 20.0,
+            'trades_per_session': 100,
+            'max_daily_trades': 500,
+            'max_position_size': Decimal('100000.00'),
+            'position_size_percentage': 10.0,
+            'default_stop_loss_percentage': 2.0,
+            'default_take_profit_percentage': 4.0,
+            'is_active': True,
+        }
+        defaults.update(kwargs)
+        
+        return cls.objects.get_or_create(
+            user_id=user_id,
+            defaults=defaults
+        )
+
+    @classmethod
+    def get_active_config(cls, user_id):
+        """Get active configuration for a user"""
+        try:
+            return cls.objects.get(user_id=user_id, is_active=True)
+        except cls.DoesNotExist:
+            return None
+
+    def calculate_position_size(self, account_balance, instrument_price):
+        """Calculate position size based on configuration and account balance"""
+        # Calculate based on percentage of capital
+        max_amount_percentage = (account_balance * self.position_size_percentage) / 100
+        
+        # Calculate based on absolute maximum
+        max_amount_absolute = float(self.max_position_size)
+        
+        # Use the smaller of the two
+        max_amount = min(max_amount_percentage, max_amount_absolute)
+        
+        # Calculate quantity (number of shares/units)
+        if instrument_price > 0:
+            quantity = int(max_amount / instrument_price)
+            return max(1, quantity)  # At least 1 unit
+        return 1
+
+    def calculate_stop_loss_price(self, entry_price, trade_direction='long'):
```

Comment:
def calculate_stop_loss_price(self, entry_price, trade_direction='long'):
 trade_management_unit/models/UserConfiguration.py
 Let's drop this method from here because this should not be placed in the model file